modules/siddhi-extensions/markov-models/src/test/java/org/wso2/siddhi/extension/markov/test/util/SiddhiTestHelper.java

```diff
@@ -0,0 +1,61 @@
+/*
+ *
+ *  * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *  *
+ *  * WSO2 Inc. licenses this file to you under the Apache License,
+ *  * Version 2.0 (the "License"); you may not use this file except
+ *  * in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing,
+ *  * software distributed under the License is distributed on an
+ *  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  * KIND, either express or implied. See the License for the
+ *  * specific language governing permissions and limitations
+ *  * under the License.
+ *
+ */
+
+package org.wso2.siddhi.extension.markov.test.util;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class SiddhiTestHelper {
```

Comment:
The usecase here is not directly translatable to a Thread synchronization problem AFAIK. There should be a thread related to Test cases and asynchronous output of Siddhi where the approach was discussed. Can you explain in more detail how the CountDownLatch can be used for this scenario. Also, will it be an over-kill for the scenario here?However, If we can apply it here, we might be able to actually change it in all the Siddhi test cases where this approach is taken. WDYT?